bdio2/src/main/java/com/blackducksoftware/bdio2/BdioEmitter.java

```diff
@@ -72,4 +78,36 @@ public void dispose() {
         }
     }
 
+    /**
+     * Check and add ScanType information for BDIO 2.0 scans the scanType value is not present but we can find it based on
+     * the publisher type.
+     *
+     * @param object
+     *            input Map which has all the values
+     */
+    @SuppressWarnings("unchecked")
+    private void checkAndAddScanType(Object object) {
+        if (object instanceof Map<?, ?>) {
+            if (!((Map<?, ?>) object).containsKey(JsonLdConsts.TYPE) || ((Map<?, ?>) object).get(JsonLdConsts.TYPE).equals("null")) {
+                if (scanType == null) {
+                    Object value = ((Map<?, ?>) object).get(Bdio.DataProperty.publisher.toString());
+                    if (value != null) {
+                        String productName = String.valueOf(((Map<?, ?>) value).get("@value"));
+                        String name = productName.substring(0, productName.indexOf('/'));
```

Comment:
Same with the product name, the product name can be extracted in the form of the Product Class that is present in the Magpie library.